src/MacVim/MMWindowController.m

```diff
@@ -110,6 +110,14 @@ - (void)_setContentHasShadow:(BOOL)shadow; // new Tiger private method
 // This is a private api that makes textured windows not have rounded corners.
 // We want this on Leopard.
 - (void)setBottomCornerRounded:(BOOL)rounded;
+
+// Private API which waits until the next frame to commit the current
+// CATransaction. Without it, a view which is set to need display will draw as
+// soon as the thread returns to the event loop; potentially drawing much more
+// frequantly than they'll appear on the screen. This method is called
+// automatically by AppKit when a *window* needs display, but not when a view
+// subtree with `wantsLayer = YES` needs display, which may be a macOS bug.
+- (void)_postWindowNeedsDisplayUnlessPostingDisabled;
```

Comment:
I'm a little nervous about adding dependencies on private APIs and prefer if we could find a way around it. @s4y can you elaborate a little more why this was added? Did it improve performance a lot? I did a brief test and didn't see a difference and wonder under what conditions you saw an improvement.

Reply:
It looks like this bug is fixed as of macOS 10.15! I updated the workaround to remove the private API dependency and only run on 10.14 and older.

On 10.14, the difference is huge when using this test:

```
let g:profstart=reltime() | for i in range(1,100) | exec "normal \<C-E>" | redraw | endfor | echo reltimestr(reltime(g:profstart)) . " seconds"
```

Without the workaround, drawing happens many times per frame:

<img width="150" alt="1.613321 seconds" src="https://user-images.githubusercontent.com/111870/67867531-edd14f00-fb00-11e9-9e96-e88f7f58b43d.png">

With the workaround, the extra draws are skipped:

<img width="150" alt="0.028653 seconds" src="https://user-images.githubusercontent.com/111870/67867604-080b2d00-fb01-11e9-891a-0f76bd287542.png">